### PR TITLE
The Witness: Small code refactor (cast_not_none)

### PR DESCRIPTION
**Pull request by NewSoupVi**
`cast`ing for not None is just so ugly, so I made a helper function for it

### CODE REVIEW

File: worlds/witness/player_items.py [hunk 1]
```diff
@@ -216,8 +216,8 @@ def get_progressive_item_ids_in_pool(self) -> Dict[int, List[int]]:
             if isinstance(item.definition, ProgressiveItemDefinition):
                 # Note: we need to reference the static table here rather than the player-specific one because the child
                 #   items were removed from the pool when we pruned out all progression items not in the settings.
-                output[cast(int, item.ap_code)] = [cast(int, static_witness_items.ITEM_DATA[child_item].ap_code)
-                                                   for child_item in item.definition.child_item_names]
+                output[cast_not_none(item.ap_code)] = [cast(int, static_witness_items.ITEM_DATA[child_item].ap_code)
```
Comment (beauxq) [inline worlds/witness/player_items.py:219]: Is this another one of the same thing inside the list comprehension?

Comment (NewSoupVi) [reply worlds/witness/player_items.py:219]: Lol yes

**Review by qwint — APPROVED** [review]
it sure is a refactor that does nothing* 
*(only kind of lying)

only read through the code (and re-read up on typing.cast() to confirm my understanding of the mechanics); did not run tests/gens/etc.

**Review by beauxq — APPROVED** [review]
I wouldn't have used the word "cast" in the function name, because `typing.cast` doesn't do the `assert`.
But it's not bad.

**Comment by NewSoupVi** [comment]
> I wouldn't have used the word "cast" in the function name, because `typing.cast` doesn't do the `assert`. But it's not bad.

That's fair, but thinking about it, it does fulfill the function of a cast conceptually (which is a more broad & cross-language concept), it just doesn't follow `typing.cast`'s specific design of a cast, so I think that's fine

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**